app/components/hooks/AppConfig/useAppConfig.tsx

```
@@ -0,0 +1,56 @@
+import { useCallback, useEffect, useState } from 'react';
+import AppConfig from './AppConfig';
+import { FetchState } from '../../../util/fetch/FetchState';
+
+type State = FetchState<AppConfig>;
+const initialState: Readonly<State> = {
+  type: 'Loading',
+};
+
+const useAppConfig = (): State => {
+  const appConfigURL =
+    'https://metamask.github.io/metamask-mobile/AppConfig/v1/AppConfig.json';
+  const [state, setState] = useState<State>(initialState);
+
+  const fetchAppConfig = useCallback(() => {
+    fetch(appConfigURL)
+      .then((response) => response.json())
+      .then((data) => {
+        try {
+          const minimumVersions = data.security.minimumVersions;
+          const appConfig: AppConfig = {
+            security: {
+              minimumVersions: {
+                appMinimumBuild: minimumVersions.appMinimumBuild,
+                appleMinimumOS: minimumVersions.appleMinimumOS,
+                androidMinimumAPIVersion:
+                  minimumVersions.androidMinimumAPIVersion,
+              },
+            },
+          };
+          setState({ type: 'Success', data: appConfig });
+        } catch (e: any) {
+          setState({
+            type: 'Error',
+            error: e,
+            message: `error parsing AppConfig ${e.message}`,
+          });
+        }
+      })
+      .catch((e: any) => {
+        setState({
```

When we get an error what is the plan at the UI level? Will it fail silent?

---

I am not sure, as you said I think the handling of errors is case by case as theoretically this could be used for a lot more data. For force upgrade I would imagine we would fail silently.

---

@sethkfman do you think I should add extra logging here or will errors be covered by sentry automatically?